### PR TITLE
Additional Blake256 support

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

**Comment by dmigwi** [comment]
@dajohi Thanks for taking time to fix this. Because of the widespread use of `btcutil.Hash160` method why don't you create another method that switches between the various supported checksum hashing functions. i.e.

```Golang
//  CkHashGen computes hash from the public key in the script input based on the
//  checksum function set.
func CkHashGen (cksumHasher base58.CksumHasher, script []byte) []byte {
    switch cksumHasher {
    case base58.Blake256D:
	return BlakeHash160(script)
    default:
	return Hash160(script)
    }
}
```

Now where `btcutil.Hash160(script)` was in use you can now replace with `btcutil.CkHashGen(p.Base58CksumHasher, script)`.

Optionally `HashI60` function could be added to the interfaces supported so that different coins can easily configure their custom implementation.

**Comment by dajohi** [comment]
I am OK with the code as is.  If @martinboehm would like a separate function, I will convert :)

**Comment by martinboehm** [comment]
I like dmigwi's suggestion. Maybe there will be another hash method to be implemented in the future.

File: address.go [hunk 1]
```diff
@@ -474,13 +468,7 @@ func (a *AddressPubKey) serialize() []byte {
...
 // Part of the Address interface.
 func (a *AddressPubKey) EncodeAddress() string {
-	var hash []byte
-	switch a.cksumHasher {
-	case base58.Blake256D:
-		hash = BlakeHash160(a.serialize())
-	default:
-		hash = Hash160(a.serialize())
-	}
+	hash := CksumHashGen(a.serialize())
```
Comment (dmigwi) [inline address.go]: Add `a.cksumHasher` parameter since `CksumHashGen` takes two parameters.

Comment (dmigwi) [reply address.go]: ~I think you already fixed this. 
Sorry I did not see it earlier.~ 
Looks like `serialize` method return values wasn't change.

**Review by dmigwi — APPROVED** [review]

**Pull request merged**